File: clipboard/ios/Plugin/ClipboardPlugin.swift
import Foundation
import Capacitor

@objc(ClipboardPlugin)
public class ClipboardPlugin: CAPPlugin {
    private let implementation = Clipboard()

    @objc func read(_ call: CAPPluginCall) {
        let result = implementation.read()

        if !result.isEmpty {
            call.resolve(result)
        } else {
            call.reject("There is no data on the clipboard")
        }
    }

    @objc func write(_ call: CAPPluginCall) {
        var success: Bool = false

        if let string = call.options["string"] as? String {
            implementation.write(content: string, ofType: Clipboard.ContentType.string)
            success = true
        } else if let urlString = call.options["url"] as? String {
            implementation.write(content: urlString, ofType: Clipboard.ContentType.url)
            success = true
        } else if let imageBase64 = call.options["image"] as? String {
            implementation.write(content: imageBase64, ofType: Clipboard.ContentType.image)
            success = true
        }

        if success {
            call.resolve()
        } else {
            call.reject("No data provided")
        }
    }
}

File: clipboard/android/src/main/java/com/capacitorjs/plugins/clipboard/ClipboardPlugin.java
package com.capacitorjs.plugins.clipboard;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Clipboard")
public class ClipboardPlugin extends Plugin {

    private Clipboard implementation;

    @Override
    public void load() {
        implementation = new Clipboard(getContext());
    }

    @PluginMethod
    public void write(PluginCall call) {
        String strVal = call.getString("string");
        String imageVal = call.getString("image");
        String urlVal = call.getString("url");
        String label = call.getString("label");

        ClipboardWriteResponse response;

        if (strVal != null) {
            response = implementation.write(label, strVal);
        } else if (imageVal != null) {
            response = implementation.write(label, imageVal);
        } else if (urlVal != null) {
            response = implementation.write(label, urlVal);
        } else {
            call.reject("No data provided");
            return;
        }

        if (response.isSuccess()) {
            call.resolve();
        } else {
            call.reject(response.getErrorMessage());
        }
    }

    @PluginMethod
    public void read(PluginCall call) {
        ClipboardData result = implementation.read();
        if (result != null) {
            JSObject ret = new JSObject();
            ret.put("value", result.getValue());
            ret.put("type", result.getType());
            call.resolve(ret);
        } else {
            call.reject("No data on clipboard");
        }
    }
}
